tests/Doctrine/Tests/ORM/UnitOfWorkTest.php

```diff
@@ -322,6 +323,34 @@ public function testRemovedAndRePersistedEntitiesAreInTheIdentityMapAndAreNotGar
     }
 
     /**
+     * @group 5579
+     */
+    public function testEntityChangeSetNotClearAfterFlushOnEntityOrArrayOfEntity()
+    {
+        // Create and Set first entity
+        $entity1 = new NotifyChangedEntity;
+        $entity1->setData('thedata');
+        $this->_unitOfWork->persist($entity1);
+
+        // Create and Set second entity
+        $entity2 = new NotifyChangedEntity;
+        $entity2->setData('thedata');
+        $this->_unitOfWork->persist($entity2);
+
+        $this->_unitOfWork->commit($entity1);
+        $this->assertCount(1, $this->_unitOfWork->getEntityChangeSet($entity2));
```

Comment:
Why is a `count` used here, instead of simply comparing the changeset? After all, we're checking for a single entry, so a `count()` operation doesn't seem to make sense to me